x/crosschain/types/cctx.go

```diff
@@ -203,6 +205,51 @@ func (m CrossChainTx) SetOutboundBallotIndex(index string) {
 }
 
 // GetCctxIndexFromBytes returns the CCTX index from a byte array.
+func (m CrossChainTx) GetValidReceiverAddress() (ethcommon.Address, error) {
+	inboundCoinType := m.InboundParams.CoinType
+	switch inboundCoinType {
+	// For coin type ZETA the receiver is added to the receiver field of the inbound vote so we can use it directly
+	case coin.CoinType_Zeta:
+		{
+			err := address.ValidateEthereumAddress(m.GetCurrentOutboundParam().Receiver)
+			if err != nil {
+				return ethcommon.Address{}, cosmoserrors.Wrap(ErrInvalidReceiverAddress, err.Error())
+			}
+			return ethcommon.HexToAddress(m.GetCurrentOutboundParam().Receiver), nil
+		}
+	// For coin type gas we create the message by composing the receiver and the call data so we need to parse the address from the RelayedMessage
+	// The receiver in this case is set to me the sender of the message
+	// For cases where this is not the case we should try and use the receiver field of the inbound vote
+	case coin.CoinType_Gas:
+		{
+			to := m.GetCurrentOutboundParam().Receiver
+			parsedAddress, _, err := chains.ParseAddressAndData(m.RelayedMessage)
+			if err != nil {
+				return ethcommon.Address{}, cosmoserrors.Wrap(ErrInvalidReceiverAddress, err.Error())
+			}
+			if parsedAddress != (ethcommon.Address{}) {
+				to = parsedAddress.String()
+			}
+			err = address.ValidateEthereumAddress(to)
+			if err != nil {
+				return ethcommon.Address{}, cosmoserrors.Wrap(ErrInvalidReceiverAddress, err.Error())
+			}
+			return ethcommon.HexToAddress(m.GetCurrentOutboundParam().Receiver), nil
+		}
+	// For coin type ERC20 we can use the receiver field of the inbound vote
+	case coin.CoinType_ERC20:
```

Comment:
This case gets the same treatment as `case coin.CoinType_Zeta`, so we could do:
```suggestion
	case coin.CoinType_ERC20, coin.CoinType_Zeta:
```